adaptations/test_adaptations.py

```
@@ -105,6 +105,23 @@ def get_rename_target_name(program, test_data):
         return "new name"
 
 
+def normalize_extension(extension):
+    if not isinstance(extension, str):
+        raise AssertionError("Extension must be a string")
```

Ruff (0.15.1)

[warning] Prefer `TypeError` exception for invalid type

(TRY004)